ArrCharOps.java

```diff
@@ -36,46 +36,81 @@ public static void println(char[] arr) {
     /** Returns the char value at the specified index. Assume that the array is non-empty.
      */
     public static char charAt(char[] arr, int index) {
-        // Replace the following statement with your code
-        return 0;
+        char returnchar = ' ';
+        for(int i =0; i<arr.length; i++)
+        {
+            if (i==index) {
+                returnchar = arr[i];
+            }
+        }
+        return returnchar;
     }
 
     /** If the two arrays have the same value in every index, 
      *  returns true; Otherwise returns false.
      */
     public static boolean equals(char[] arr1, char[] arr2) {
-        // Replace the following statement with your code
+        int counter = 0;
+        for(int i=0; i< arr1.length; i++)
+        {
+            if (arr1[i]==arr2[i]) {
+                counter++;
+            }
+        }
+        if (counter==arr1.length) {
+            return true;
```

Comment:
The following is incorrect code.. 
Let me give you a couple of examples. 
for convenience I will showcase my examples with strings but could be applied to char arrays.
1. Example 1: 
`str1 (arr1) = "hell", str2 (arr2) = "hello" `
from the get go, we can tell that expected output will be `false`

Your code will go over array and since first 4 chars will be matching hence, counter's value after loop end will result in value `4` since str1.length() /* arr.length */ is 4 hence will return `true`

Example 2: 
`str1 (arr1) = "hello", str2 (arr2) = "hell"`
from the get go, we can tell that expected output will be `false`

your code will run smoothly, up to last iteration. 
in last iteration, the code will throw an `IndexOutOfBoundsException`

The solution is to add a check prior to loop which verifies if lengths of arrays are equal or not.. 
`if (arr1.length != arr2.length)`

A similar equals method have been implemented in recitation 4. 
please go over it and contact me via email / visit office hours for further explanation.